src/platform/packages/shared/kbn-scout/src/cli/config_discovery.test.ts

```diff
@@ -0,0 +1,161 @@
+  const mockPluginsWithConfigs = new Map([
+    ['pluginA', { group: 'groupA', configs: ['config1.ts', 'config2.ts'] }], // enabled
+    ['pluginB', { group: 'groupB', configs: ['config3.ts'] }], // disabled
+  ]);
+
+  beforeAll(() => {
+    flagsReader = {
+      arrayOfStrings: jest.fn(),
+      boolean: jest.fn(),
+    } as any;
+
+    log = {
+      info: jest.fn(),
+      error: jest.fn(),
+      warn: jest.fn(),
+    } as any;
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    (fs.existsSync as jest.Mock).mockReturnValue(false);
+    (fs.mkdirSync as jest.Mock).mockImplementation(jest.fn());
+    (fs.writeFileSync as jest.Mock).mockImplementation(jest.fn());
+
+    (measurePerformance as jest.Mock).mockImplementation((_log, _msg, fn) => fn());
+
+    (getScoutPlaywrightConfigs as jest.Mock).mockReturnValue(mockPluginsWithConfigs);
+    (validateWithScoutCiConfig as jest.Mock).mockReturnValue(mockPluginsWithRunnableConfigs);
+  });
+
+  it('validates configs when "validate" is true', () => {
+    // force --validate
+    flagsReader.boolean.mockImplementation((flag) => (flag === 'validate' ? true : false));
+
+    runDiscoverPlaywrightConfigs(flagsReader, log);
+
+    // setting --validate will trigger a validation
+    expect(validateWithScoutCiConfig).toHaveBeenCalledWith(log, mockPluginsWithConfigs);
+  });
+
+  it('should correctly parse custom config search paths', () => {
+    flagsReader.arrayOfStrings.mockReturnValue(['customConfigSearchPaths']);
+    runDiscoverPlaywrightConfigs(flagsReader, log);
+    expect(getScoutPlaywrightConfigs).toHaveBeenCalledWith(['customConfigSearchPaths'], log);
+  });
+
+  it('logs found configs when they exist and "save" flag is false', () => {
+    // never --save
+    flagsReader.boolean.mockImplementation((flag) => (flag === 'save' ? false : false));
+
+    runDiscoverPlaywrightConfigs(flagsReader, log);
+
+    expect(log.info.mock.calls).toEqual([
+      ["Found Playwright config files in '2' plugins"],
+      ['groupA / [pluginA] plugin:'],
+      ['- config1.ts'],
+      ['- config2.ts'],
+      ['groupB / [pluginB] plugin:'],
+      ['- config3.ts'],
+    ]);
+  });
+
+  it('logs "No Playwright config files found" when no configs are found', () => {
+    flagsReader.arrayOfStrings.mockReturnValue([]);
+    flagsReader.boolean.mockReturnValue(false);
+
+    (getScoutPlaywrightConfigs as jest.Mock).mockReturnValue(new Map());
+    runDiscoverPlaywrightConfigs(flagsReader, log);
+
+    expect(log.info).toHaveBeenCalledWith('No Playwright config files found');
+  });
+
+  it('validates and saves enabled plugins with their config files when --save is set', () => {
+    // force --save
+    flagsReader.boolean.mockImplementation((flag) => {
+      if (flag === 'save') {
+        return true;
+      }
+
+      if (flag === 'validate') {
+        return false;
+      }
+
+      return false;
+    });
```

Comment:
nice! I think this is the way to go in most of the tests: it will save us time if when we add more flags in the future